src/main/java/com/odysseusinc/arachne/executionengine/service/impl/KerberosServiceImpl.java

```diff
@@ -84,15 +107,50 @@ public void kinit(DataSourceUnsecuredDTO dataSource, File workDir) throws IOExce
                 } catch (InterruptedException e) {
                     log.error("Failed to obtain kerberos ticket", e);
                 }
-            }finally {
+            } finally {
                 if (Objects.nonNull(keytab)) {
                     FileUtils.deleteQuietly(keytab.toFile());
                 }
             }
         }
     }
 
-    private void klist(File workDir){
+    private void addKrbRealmToConfig(DataSourceUnsecuredDTO dataSource) throws IOException {
+
+        Context context = new Context();
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("realmName", dataSource.getKrbRealm());
+        parameters.put("adminServer", dataSource.getKrbFQDN());
+        parameters.put("kdcServer", dataSource.getKrbFQDN());
+        context.setVariables(parameters);
+        String htmlString = templateEngine.process(templateName, context);
+
+        Source source = new Source(htmlString);
+        String textString = source.getRenderer().toString();
+
+        File config = new File(configPath);
+        long fileLength = config.length();
+        try (RandomAccessFile raf = new RandomAccessFile(config, "rw"); FileChannel channel = raf.getChannel()) {
```

Comment:
Potentially [realms] might be not the latest section of krb5.conf, so simply adding realm to the end of file is dangerous.
Why not to add line just under the [realms] ?
And what about concurrency with RandomAccessFile?